examples/with-mdx/package.json

```diff
@@ -6,8 +6,7 @@
     "start": "next start"
   },
   "dependencies": {
-    "@mdx-js/loader": "^1.5.1",
-    "@mdx-js/react": "^1.6.18",
```

Comment:
There's a separate example for app router, should this dep remain?

https://github.com/vercel/next.js/tree/canary/examples/app-dir-mdx

Reply:
True, it's being covered there.

Reply:
The benefit that `@mdx-js/react` offers, is that it allows users to provide components for MDX files for different files in the React tree. IMO this is a niche use case. I think the `mdx-components` file is a much better solution for Next.js. Kudos to whoever came up with that.

`@mdx-js/react` works for the pages directory, but not for the app directory. It just crashes with a weird error. I personally found this confusing and I’ve heard so from others too.

Regardless of what works or doesn’t work, IMO it’s good to have a single blessed way to use MDX with Next.js. I think it’s hard to explain what the type of router has to do with this.

Using the `mdx-components` file better aligns with the DX after I implement [support for provided components](https://github.com/mdx-js/mdx-analyzer/issues/399) in the MDX language server. Just define them once globally, then use them in every MDX file.

What is missing from this PR though, is the addition of the `mdx-components` file. I could add it, but even better would be to merge #59693.

Reply:
FYI: The `mdx-components` file is only for the App Router, not for the Pages Router.

Reply:
Interesting... good point! TIL